pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -40,9 +39,6 @@ func (t *Task) Validate(ctx context.Context) *apis.FieldError {
 }
 
 func (ts *TaskSpec) Validate(ctx context.Context) *apis.FieldError {
-	if equality.Semantic.DeepEqual(ts, &TaskSpec{}) {
-		return apis.ErrMissingField(apis.CurrentField)
-	}
 
 	if len(ts.Steps) == 0 {
```

Comment:
What will happen if ts is a nil pointer? Is there protection against that elsewhere in the codebase that will prevent it from happening?

Reply:
The way the code is now, the callers of the  validate methods appear to be responsible for checking for nil.  For example in taskrun_validation:

```
	// Validate TaskSpec if it's present
	if ts.TaskSpec != nil {
		if err := ts.TaskSpec.Validate(ctx); err != nil {
			return err
		}
	}
```
I haven't seen a nil check inside any of the validation methods I've looked at.

Reply:
I'm not sure this is always the case though - see Task.Validate a few lines above this one:

```go
func (t *Task) Validate(ctx context.Context) *apis.FieldError {
	if err := validate.ObjectMetadata(t.GetObjectMeta()); err != nil {
		return err.ViaField("metadata")
	}
	return t.Spec.Validate(ctx)
}
```

I'm fine to remove the check in TaskSpec.Validate if we're absolutely sure that there won't be usages.  But if we're cherry-picking this into the 0.11 branch just before we cut the beta and we're not 100% sure then I'd prefer we postpone this change to another PR.  WDYT?

Reply:
(Also, I believe - though I could be completely wrong! - that `equality.Semantic.DeepEqual` will actually consider a nil pointer and &TaskSpec{} as semantically equal, so this would return an error if there was a nil pointer.  We should have unit testing around this.)

Reply:
No, the first thing I did was to check if the DeepEqual check was catching a nil.  I put the check back and forced ts to be nil.  It fell through the DeepEqual check and panicked on the ts.Steps reference.  So I don't believe I'm introducing an exposure.  It was there before.

Reply:
(fwiw I don't think this is fixing an urgent problem so it's fine to postpone merging it to post 0.11)

Reply:
👍 understood, thanks for testing the nil case and correcting my assumption here.